pkg/internal/bundle.go

```diff
@@ -0,0 +1,134 @@
+package manifests
+
+import (
+	"encoding/json"
+
+	"github.com/blang/semver"
+	operatorsv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/operator-framework/operator-registry/pkg/sqlite"
+	"github.com/pkg/errors"
+)
+
+// TODO: use internal version of registry.Bundle/registry.PackageManifest so
+// operator-registry can use validation library.
+
+// manifestsLoad loads a manifests directory from disk.
+type manifestsLoad struct {
+	dir     string
+	pkg     registry.PackageManifest
+	bundles map[string]*registry.Bundle
+}
+
+// Ensure manifestsLoad implements registry.Load.
+var _ registry.Load = &manifestsLoad{}
+
+// populate uses operator-registry's sqlite.NewSQLLoaderForDirectory to load
+// l.dir's manifests. Note that this method does not call any functions that
+// use SQL drivers.
+func (l *manifestsLoad) populate() error {
+	loader := sqlite.NewSQLLoaderForDirectory(l, l.dir)
+	if err := loader.Populate(); err != nil {
+		return errors.Wrapf(err, "error getting bundles from manifests dir %q", l.dir)
+	}
+	return nil
+}
+
+// AddOperatorBundle adds a bundle to l.
+func (l *manifestsLoad) AddOperatorBundle(bundle *registry.Bundle) error {
+	csvRaw, err := bundle.ClusterServiceVersion()
+	if err != nil {
+		return errors.Wrap(err, "error getting bundle CSV")
+	}
+	csvSpec := operatorsv1alpha1.ClusterServiceVersionSpec{}
+	if err := json.Unmarshal(csvRaw.Spec, &csvSpec); err != nil {
+		return errors.Wrap(err, "error unmarshaling CSV spec")
+	}
+	bundle.Name = csvSpec.Version.String()
+	l.bundles[csvSpec.Version.String()] = bundle
+	return nil
+}
+
+// AddOperatorBundle adds the package manifest to l.
+func (l *manifestsLoad) AddPackageChannels(pkg registry.PackageManifest) error {
+	l.pkg = pkg
+	return nil
+}
+
+// AddBundlePackageChannels is a no-op to implement the registry.Load interface.
+func (l *manifestsLoad) AddBundlePackageChannels(manifest registry.PackageManifest, bundle registry.Bundle) error {
```

Comment:
Note to self: As we change operator-registry we'll need to be aware of the dependency here on the implementation of its Load interface.

Reply:
In general we're going to have to do something about operator-registry types. If we want this to be a dependency of operator-registry, we're either going to have to copy over types or copy over parse code and use internal types. WDYT? We can probably clean this up in a follow-up PR.